tests/transforms/test_decompositions.py

```diff
@@ -855,3 +855,93 @@ def test_two_qubit_decomposition_tensor_products_jax(self, U_pair, wires):
         obtained_matrix = qml.matrix(tape, wire_order=wires)
 
         assert check_matrix_equivalence(U, obtained_matrix, atol=1e-7)
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("wires", [[0, 1], ["a", "b"], [3, 2], ["c", 0]])
+    @pytest.mark.parametrize("U", samples_3_cnots + samples_2_cnots + samples_1_cnot)
+    def test_two_qubit_decomposition_jax_jit(self, U, wires):
+        """Test that the results of decomposing a two-qubit operation in JAX with and without jitting match."""
+        import jax
+        from jax.config import config
+
+        remember = config.read("jax_enable_x64")
+        config.update("jax_enable_x64", True)
+
+        U = jax.numpy.array(U, dtype=jax.numpy.complex128)
+
+        dev = qml.device("default.qubit", wires=wires)
+
+        @jax.jit
+        @qml.qnode(dev, interface="jax")
+        def f_jit(U):
+            for g in qml.QubitUnitary(U, wires=wires, do_queue=False).decomposition():
+                qml.apply(g)
+            return (
+                qml.expval(qml.PauliX(wires[0])),
+                qml.expval(qml.PauliY(wires[0])),
+                qml.expval(qml.PauliZ(wires[0])),
+                qml.expval(qml.PauliX(wires[1])),
+                qml.expval(qml.PauliY(wires[1])),
+                qml.expval(qml.PauliZ(wires[1])),
+            )
```

Comment:
I think it might be worth adding `qml.state` at the end of the QNode or checking if the unitary made up of the decomposition matches (up to a global phase). Expectation values could match even when two unitaries evolve the state in considerably different ways.

Reply:
I agree with @antalszava here.

Reply:
It now checks matrix equivalency like the other tests